### DISCUSSION TITLE
SmoothArc on GC9A01

### DISCUSSION THREAD

**Discussion by pink-fr3ud** [Bodmer/TFT_eSPI#2399] [Q&A - General]
Hi,
I'm using with success this library for some time know, and I came across a problem with the new drawSmoothArc function, rounded ends are not that smooth. Disabling rounded ends makes a crisp end. I tried using a even or uneven number for the thickness of the arc, but same same.
![IMG_5155](https://user-images.githubusercontent.com/40762492/218329234-ababaab8-cb03-4d61-8dde-22a8150d2973.jpg)

Config: ESP32 with GC9A01
`  tft.drawSmoothArc(tft.width()/2, tft.height()/2, 110, 80, 120, 240, fg_color, bg_color, true);`

Thank you for the hardwork

**Comment by Bodmer** [comment]
I have investigated this and it appears the GC9A01 has a pixel stream time-out that stops new pixels in a started stream from being rendered.  This is a behaviour only relevant to that particular display it seems. I have raised this as an Issue.

**Comment by Bodmer** [comment]
I have updated the master library and this is fixed now. Thanks for reporting it.

**Comment by pink-fr3ud** [comment]
You are insane, I took more time to test the fix than you finding the problem and fixing it
It works like a charm no more glitches !
Thank you again

**Comment by datort** [comment]
Hey @pink-fr3ud, 

did the fix work for you? I'm experiencing the same issue using the lib in version 2.5.0. 🤔 

Thanks

**Reply by pink-fr3ud** [reply]
Yes it worked !

**Reply by datort** [reply]
Ok, my bad. Delivered the answer to the cause in my question: As bodmer wrote, the fix is in the master but not yet in a released version. So if anyone else runs into this while installing the lib via Arduino IDE: the fix is in the master branch and will be available in 2.5.1 or what ever the next version will be. Thanks for your answer and to bodmer for the fix.
